### ISSUE TITLE
DH X9.42 DER serialization

### ISSUE THREAD

**Issue by palaviv** [openssl/openssl#3102]
I have the following PEM X9.42 DH parameters:
```
-----BEGIN X9.42 DH PARAMETERS-----
MIICKQKCAQEArRB+HpEjqdDWYPqnlVnFH6INZOVoO5/RtUsVl7YdCnXm+hQd+VpW
26+aPEB7od8V6z1oijCcGA4d5rhaEnSgpm0/gVKtasISkDfJ7e/aTfjZHo/vVbc5
S3rVt9C2wSIHyfmNEe002/bGugssi7wnvmoA4KC5xJcIs7+KMXCRiDaBKGEwvImF
2xYC5xRBXZMwJ4Jzx94x79xzEPcSH9WgdBWYfZrcCkhtzfk6zEQyg4cxXXXhmMZB
pIDNhqG55YfovmDmnMkosrnFIXLkEwQumyPxCw4W55djybU9z0uoCinj+3PBa451
uX7zY+L/ox9xz53lOE5xuBwKxN/+DBDmTwKCAQEArEAy708tmuOd8wtcj/2sUGze
vnuJmYyvdIZqCM/k/+OmgkpOELmm8N2SHwGnDEr6q3OddwDCn1LFfbF8YgqGUr5e
kAGo1mrXwXZpEBmZAkr00CcnWsE0i7inYtBSG8mK4kcVBCLqHtQJk51U2nRgzbX2
xrJQcXy+8YDrNBGOmNEZUppF1vg0Vm4wJeMWozDvu3eobwwasVsFGuPUKMj4rLcK
gTcVC47rEOGD7dGZY93Z4mPkdwWJ72qiHn9fL/OBtTnM40CdE81Wavu0jWwBkYHh
vP6UswJp7f5y/ptqpL17Wg8ccc//TBnEGOH27AF5gbwIfypwZbOEuJDTGR8r+gId
AIAcDTTFjZP+mXF3EB+AU1pHOM68vziambNjces=
-----END X9.42 DH PARAMETERS-----
```
I wish to convert this PEM to DER format. I am trying to do `openssl dhparam -in dhp_rfc5114_2.pem -inform pem -outform DER > /tmp/dhp_rfc5114_2.der` but when reading the created DER the subgroup order seems to disappear.

`openssl dhparam -in /tmp/dhp_rfc5114_2.der -inform DER -text -noout`:
```
DH Parameters: (2048 bit)
        prime:
            00:ad:10:7e:1e:91:23:a9:d0:d6:60:fa:a7:95:59:
            c5:1f:a2:0d:64:e5:68:3b:9f:d1:b5:4b:15:97:b6:
            1d:0a:75:e6:fa:14:1d:f9:5a:56:db:af:9a:3c:40:
            7b:a1:df:15:eb:3d:68:8a:30:9c:18:0e:1d:e6:b8:
            5a:12:74:a0:a6:6d:3f:81:52:ad:6a:c2:12:90:37:
            c9:ed:ef:da:4d:f8:d9:1e:8f:ef:55:b7:39:4b:7a:
            d5:b7:d0:b6:c1:22:07:c9:f9:8d:11:ed:34:db:f6:
            c6:ba:0b:2c:8b:bc:27:be:6a:00:e0:a0:b9:c4:97:
            08:b3:bf:8a:31:70:91:88:36:81:28:61:30:bc:89:
            85:db:16:02:e7:14:41:5d:93:30:27:82:73:c7:de:
            31:ef:dc:73:10:f7:12:1f:d5:a0:74:15:98:7d:9a:
            dc:0a:48:6d:cd:f9:3a:cc:44:32:83:87:31:5d:75:
            e1:98:c6:41:a4:80:cd:86:a1:b9:e5:87:e8:be:60:
            e6:9c:c9:28:b2:b9:c5:21:72:e4:13:04:2e:9b:23:
            f1:0b:0e:16:e7:97:63:c9:b5:3d:cf:4b:a8:0a:29:
            e3:fb:73:c1:6b:8e:75:b9:7e:f3:63:e2:ff:a3:1f:
            71:cf:9d:e5:38:4e:71:b8:1c:0a:c4:df:fe:0c:10:
            e6:4f
        generator:
            00:ac:40:32:ef:4f:2d:9a:e3:9d:f3:0b:5c:8f:fd:
            ac:50:6c:de:be:7b:89:99:8c:af:74:86:6a:08:cf:
            e4:ff:e3:a6:82:4a:4e:10:b9:a6:f0:dd:92:1f:01:
            a7:0c:4a:fa:ab:73:9d:77:00:c2:9f:52:c5:7d:b1:
            7c:62:0a:86:52:be:5e:90:01:a8:d6:6a:d7:c1:76:
            69:10:19:99:02:4a:f4:d0:27:27:5a:c1:34:8b:b8:
            a7:62:d0:52:1b:c9:8a:e2:47:15:04:22:ea:1e:d4:
            09:93:9d:54:da:74:60:cd:b5:f6:c6:b2:50:71:7c:
            be:f1:80:eb:34:11:8e:98:d1:19:52:9a:45:d6:f8:
            34:56:6e:30:25:e3:16:a3:30:ef:bb:77:a8:6f:0c:
            1a:b1:5b:05:1a:e3:d4:28:c8:f8:ac:b7:0a:81:37:
            15:0b:8e:eb:10:e1:83:ed:d1:99:63:dd:d9:e2:63:
            e4:77:05:89:ef:6a:a2:1e:7f:5f:2f:f3:81:b5:39:
            cc:e3:40:9d:13:cd:56:6a:fb:b4:8d:6c:01:91:81:
            e1:bc:fe:94:b3:02:69:ed:fe:72:fe:9b:6a:a4:bd:
            7b:5a:0f:1c:71:cf:ff:4c:19:c4:18:e1:f6:ec:01:
            79:81:bc:08:7f:2a:70:65:b3:84:b8:90:d3:19:1f:
            2b:fa
```
 While `openssl dhparam -in vectors/cryptography_vectors/asymmetric/DH/dhp_rfc5114_2.pem -inform pem -text -noout`:
```
DH Parameters: (2048 bit)
        prime:
            00:ad:10:7e:1e:91:23:a9:d0:d6:60:fa:a7:95:59:
            c5:1f:a2:0d:64:e5:68:3b:9f:d1:b5:4b:15:97:b6:
            1d:0a:75:e6:fa:14:1d:f9:5a:56:db:af:9a:3c:40:
            7b:a1:df:15:eb:3d:68:8a:30:9c:18:0e:1d:e6:b8:
            5a:12:74:a0:a6:6d:3f:81:52:ad:6a:c2:12:90:37:
            c9:ed:ef:da:4d:f8:d9:1e:8f:ef:55:b7:39:4b:7a:
            d5:b7:d0:b6:c1:22:07:c9:f9:8d:11:ed:34:db:f6:
            c6:ba:0b:2c:8b:bc:27:be:6a:00:e0:a0:b9:c4:97:
            08:b3:bf:8a:31:70:91:88:36:81:28:61:30:bc:89:
            85:db:16:02:e7:14:41:5d:93:30:27:82:73:c7:de:
            31:ef:dc:73:10:f7:12:1f:d5:a0:74:15:98:7d:9a:
            dc:0a:48:6d:cd:f9:3a:cc:44:32:83:87:31:5d:75:
            e1:98:c6:41:a4:80:cd:86:a1:b9:e5:87:e8:be:60:
            e6:9c:c9:28:b2:b9:c5:21:72:e4:13:04:2e:9b:23:
            f1:0b:0e:16:e7:97:63:c9:b5:3d:cf:4b:a8:0a:29:
            e3:fb:73:c1:6b:8e:75:b9:7e:f3:63:e2:ff:a3:1f:
            71:cf:9d:e5:38:4e:71:b8:1c:0a:c4:df:fe:0c:10:
            e6:4f
        generator:
            00:ac:40:32:ef:4f:2d:9a:e3:9d:f3:0b:5c:8f:fd:
            ac:50:6c:de:be:7b:89:99:8c:af:74:86:6a:08:cf:
            e4:ff:e3:a6:82:4a:4e:10:b9:a6:f0:dd:92:1f:01:
            a7:0c:4a:fa:ab:73:9d:77:00:c2:9f:52:c5:7d:b1:
            7c:62:0a:86:52:be:5e:90:01:a8:d6:6a:d7:c1:76:
            69:10:19:99:02:4a:f4:d0:27:27:5a:c1:34:8b:b8:
            a7:62:d0:52:1b:c9:8a:e2:47:15:04:22:ea:1e:d4:
            09:93:9d:54:da:74:60:cd:b5:f6:c6:b2:50:71:7c:
            be:f1:80:eb:34:11:8e:98:d1:19:52:9a:45:d6:f8:
            34:56:6e:30:25:e3:16:a3:30:ef:bb:77:a8:6f:0c:
            1a:b1:5b:05:1a:e3:d4:28:c8:f8:ac:b7:0a:81:37:
            15:0b:8e:eb:10:e1:83:ed:d1:99:63:dd:d9:e2:63:
            e4:77:05:89:ef:6a:a2:1e:7f:5f:2f:f3:81:b5:39:
            cc:e3:40:9d:13:cd:56:6a:fb:b4:8d:6c:01:91:81:
            e1:bc:fe:94:b3:02:69:ed:fe:72:fe:9b:6a:a4:bd:
            7b:5a:0f:1c:71:cf:ff:4c:19:c4:18:e1:f6:ec:01:
            79:81:bc:08:7f:2a:70:65:b3:84:b8:90:d3:19:1f:
            2b:fa
        subgroup order:
            00:80:1c:0d:34:c5:8d:93:fe:99:71:77:10:1f:80:
            53:5a:47:38:ce:bc:bf:38:9a:99:b3:63:71:eb
```

**Comment by mattcaswell** [comment]
This appears to be a bug. Fix in #3111 (for master/1.1.0) and #3112 (for 1.0.2).

**Comment by mattcaswell** [comment]
Fix has been pushed to all branches.
